### PR TITLE
Std xml utils

### PR DESCRIPTION
**Pull request by NotLebedev**
# Description

Add `xaccess`,`xupdate` and `xinsert` scripts to standard library. They allow accessing and manipulating data in new xml format https://github.com/nushell/nushell/pull/7947 with relative ease.

Access some data in nushell xml structure:
image

Update attributes of xml tags matching a path:
image

# User-Facing Changes

New commands `std xaccess`, `std xupdate` and `std xinsert`

# Tests + Formatting

Don't forget to add tests that cover your changes.

Make sure you've run and fixed any issues with these commands:

- `cargo fmt --all -- --check` to check standard code formatting (`cargo fmt --all` applies these changes)
- `cargo clippy --workspace -- -D warnings -D clippy::unwrap_used -A clippy::needless_collect` to check that you're using the standard code style
- `cargo test --workspace` to check that all tests pass

> **Note**
> from `nushell` you can also use the `toolkit` as follows
> ```bash
> use toolkit.nu  # or use an `env_change` hook to activate it automatically
> toolkit check pr
> ```

# After Submitting

If your PR had any user-facing changes, update [the documentation](https://github.com/nushell/nushell.github.io) after the PR is merged, if necessary. This will help us keep the docs up to date.

### CODE REVIEW

**Comment by NotLebedev** [comment]
Maybe we should ship this in upcoming release, otherwise there will be exactly 0 ways to work with new xml format in a sane way. This is very bare bones, but still usefulel, and I plan to add more utilities and better this two.

**Comment by amtoine** [comment]
thanks for your contribution! :wave: 

the thing is i do not know XML at all, so i do not really know what to test and what behaviour to look after :open_mouth: 

for instance
```bash
open Cargo.toml | to xml
```
does not even work :laughing: 

in the meantime, i'll have a look at the source (but it looks already quite good and well documented, i like that :smirk:)

**Review by amtoine — CHANGES_REQUESTED** [review]
i really do not understand the XML format, that's even the first time i see it for real i think :open_mouth: 

so i'll trust you and other XML-experts that the new `xml` module does what it's supposed to do :relieved: 

only minor remarks and i'm wondering if aligning the right operands of the `assert`s would make them a bit easier to read?
i do not know, as you feel is the best.

**Comment by NotLebedev** [comment]
Finished this. Fixed all recommendations, added one more useful command.

Regarding 

> the thing is i do not know XML at all, so i do not really know what to test and what behaviour to look after 😮
> 
> for instance
> 
> ```shell
> open Cargo.toml | to xml
> ```
> 
> does not even work 😆

this is the reason why this commands are required at all. Xml is a weird format among others because it is not a tabular or key-value format (which are easy to use in nushell) but a strange tag-attributes-content model, allowing repetitions of entries but placing many other arbitrary restrictions. This is why `to xml` and `from xml` commands accept and produce very specific format, consisting of `record<tag attribute content>`. The help  for `to xml` and `from xml` describes it in more detail.


TL;DR xml does not map on json-like formats well, thus it can not be mapped one-to one with most nushell values in a sensible way.

**Comment by amtoine** [comment]
@NotLebedev 

thanks for the explaination, much appreciate it :heart: 

i'm gonna review right now as it's ready :muscle:

File: crates/nu-utils/standard_library/tests.nu [hunk 1]
```diff
@@ -80,7 +80,6 @@ def main [
         | flatten
         | rename file module name
     )
-
```
Comment (amtoine) [inline crates/nu-utils/standard_library/tests.nu:83]: coul you put that line back, for readability of the test runner? :yum:

File: test.nu
Comment (amtoine) [inline test.nu:1]: this file does not look right :thinking: 

looks to me it should be removed, like from some intermediate test.

File: crates/nu-utils/standard_library/test_xml.nu
Comment (amtoine) [inline crates/nu-utils/standard_library/test_xml.nu:1]: the diff of that file looks super strange, i'm not sure why it has to remove all the previous `std.nu`, put it back and then add your work :thinking:

**Review by amtoine — CHANGES_REQUESTED** [review]
a few little things, we're close :yum:

File: crates/nu-utils/standard_library/test_xml.nu
Comment (NotLebedev) [reply crates/nu-utils/standard_library/test_xml.nu:1]: I work on windows and this is the crlf bullshit that comes with it. Not sure what happened that this file converted mid-work.

**Comment by fdncred** [comment]
@NotLebedev can we resolve the conflicts so we can land this please?

**Comment by fdncred** [comment]
Thanks!

**Comment by fdncred** [comment]
@NotLebedev any ideas why I'm getting errors with `xaccess` on linux?
image

**Comment by fdncred** [comment]
Sorry @NotLebedev - it was my fault. i had another `match` command defined.
